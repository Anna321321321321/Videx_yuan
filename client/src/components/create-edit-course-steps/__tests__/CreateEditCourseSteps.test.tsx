import { shallow } from 'enzyme';
import moment from 'moment';
import React from 'react';
import CreateEditCourseSteps from '../';

describe('Create Edit Course Steps Unit Tests: Create', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CreateEditCourseSteps mode="create" />);
  });

  it('Step 0', () => {
    expect(wrapper.find('Step').length).toBe(3);
    expect(wrapper).toMatchSnapshot();
  });

  it('Step 1', () => {
    wrapper.setState({
      fields: {
        name: {
          value: 'test'
        },
        releaseDate: {
          value: moment('1970-01-01 16:00Z')
        }
      }
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('Step 2: No Loading', () => {
    wrapper.setState({
      current: 1,
      fields: {
        name: {
          value: 'test'
        },
        releaseDate: {
          value: moment('1970-01-01 16:00Z')
        }
      }
    });
    expect(wrapper.find('Button').length).toBe(2);
    expect(
      wrapper
        .find('Button')
        .at(0)
        .prop('loading')
    ).toBe(false);
    expect(wrapper).toMatchSnapshot();
  });

  it('Step 2: Loading', () => {
    wrapper.setState({
      current: 1,
      fields: {
        name: {
          value: 'test'
        },
        releaseDate: {
          value: moment('1970-01-01 16:00Z')
        }
      },
      loading: true
    });
    expect(wrapper.find('Button').length).toBe(2);
    expect(
      wrapper
        .find('Button')
        .at(0)
        .prop('loading')
    ).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('Step 3: No Error', () => {
    wrapper.setState({
      current: 2,
      fields: {
        name: {
          value: 'test'
        },
        releaseDate: {
          value: moment('1970-01-01 16:00Z')
        }
      },
      status: 'finish'
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('Step 3: Error', () => {
    wrapper.setState({
      current: 2,
      fields: {
        name: {
          value: 'test'
        },
        releaseDate: {
          value: moment('1970-01-01 16:00Z')
        }
      },
      status: 'error'
    });
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Create Edit Course Steps Unit Tests: Edit', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <CreateEditCourseSteps
        mode="edit"
        id="id"
        name="name"
        releaseDate="1970-01-01 16:00Z"
      />
    );
  });

  it('Check States', () => {
    expect(wrapper.state('fields')).toEqual({
      name: {
        value: 'name'
      },
      releaseDate: {
        value: moment('1970-01-01 16:00Z')
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
