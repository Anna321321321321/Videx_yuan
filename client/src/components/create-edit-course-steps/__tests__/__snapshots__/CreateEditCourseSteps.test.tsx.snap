// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Edit Course Steps Unit Tests: Create Step 0 1`] = `
<Adapter
  style={
    Object {
      "padding": "0px 24px",
    }
  }
>
  <div
    style={
      Object {
        "background": "#fff",
        "padding": "24px",
      }
    }
  >
    <Steps
      current={0}
      iconPrefix="ant"
      prefixCls="ant-steps"
      status="wait"
    >
      <Step
        key="Configure"
        title="Configure"
      />
      <Step
        key="Review"
        title="Review"
      />
      <Step
        key="Finished"
        title="Finished"
      />
    </Steps>
    <div
      style={
        Object {
          "backgroundColor": "#fafafa",
          "border": "1px dashed #e9e9e9",
          "borderRadius": "6px",
          "marginTop": "24px",
          "minHeight": "200px",
          "paddingTop": "72px",
          "textAlign": "center",
        }
      }
    >
      <Form(CreateEditCourseForm)
        name={
          Object {
            "value": null,
          }
        }
        onChange={[Function]}
        onSubmit={[Function]}
        releaseDate={
          Object {
            "value": null,
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "marginTop": "24px",
        }
      }
    />
  </div>
</Adapter>
`;

exports[`Create Edit Course Steps Unit Tests: Create Step 1 1`] = `
<Adapter
  style={
    Object {
      "padding": "0px 24px",
    }
  }
>
  <div
    style={
      Object {
        "background": "#fff",
        "padding": "24px",
      }
    }
  >
    <Steps
      current={0}
      iconPrefix="ant"
      prefixCls="ant-steps"
      status="wait"
    >
      <Step
        key="Configure"
        title="Configure"
      />
      <Step
        key="Review"
        title="Review"
      />
      <Step
        key="Finished"
        title="Finished"
      />
    </Steps>
    <div
      style={
        Object {
          "backgroundColor": "#fafafa",
          "border": "1px dashed #e9e9e9",
          "borderRadius": "6px",
          "marginTop": "24px",
          "minHeight": "200px",
          "paddingTop": "72px",
          "textAlign": "center",
        }
      }
    >
      <Form(CreateEditCourseForm)
        name={
          Object {
            "value": "test",
          }
        }
        onChange={[Function]}
        onSubmit={[Function]}
        releaseDate={
          Object {
            "value": "1970-01-01T16:00:00.000Z",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "marginTop": "24px",
        }
      }
    />
  </div>
</Adapter>
`;

exports[`Create Edit Course Steps Unit Tests: Create Step 2: Loading 1`] = `
<Adapter
  style={
    Object {
      "padding": "0px 24px",
    }
  }
>
  <div
    style={
      Object {
        "background": "#fff",
        "padding": "24px",
      }
    }
  >
    <Steps
      current={1}
      iconPrefix="ant"
      prefixCls="ant-steps"
      status="wait"
    >
      <Step
        key="Configure"
        title="Configure"
      />
      <Step
        key="Review"
        title="Review"
      />
      <Step
        key="Finished"
        title="Finished"
      />
    </Steps>
    <div
      style={
        Object {
          "backgroundColor": "#fafafa",
          "border": "1px dashed #e9e9e9",
          "borderRadius": "6px",
          "marginTop": "24px",
          "minHeight": "200px",
          "paddingTop": "72px",
          "textAlign": "center",
        }
      }
    >
      <div>
        <p>
          Name: 
          test
        </p>
        <p>
          Start releaseDate:
           
          1970-01-01
        </p>
      </div>
    </div>
    <div
      style={
        Object {
          "marginTop": "24px",
        }
      }
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
          }
        }
      >
        <Button
          block={false}
          ghost={false}
          loading={true}
          onClick={[Function]}
          prefixCls="ant-btn"
          type="primary"
        >
          Submit
        </Button>
        <Button
          block={false}
          ghost={false}
          loading={false}
          onClick={[Function]}
          prefixCls="ant-btn"
          style={
            Object {
              "marginLeft": 8,
            }
          }
        >
          Previous
        </Button>
      </div>
    </div>
  </div>
</Adapter>
`;

exports[`Create Edit Course Steps Unit Tests: Create Step 2: No Loading 1`] = `
<Adapter
  style={
    Object {
      "padding": "0px 24px",
    }
  }
>
  <div
    style={
      Object {
        "background": "#fff",
        "padding": "24px",
      }
    }
  >
    <Steps
      current={1}
      iconPrefix="ant"
      prefixCls="ant-steps"
      status="wait"
    >
      <Step
        key="Configure"
        title="Configure"
      />
      <Step
        key="Review"
        title="Review"
      />
      <Step
        key="Finished"
        title="Finished"
      />
    </Steps>
    <div
      style={
        Object {
          "backgroundColor": "#fafafa",
          "border": "1px dashed #e9e9e9",
          "borderRadius": "6px",
          "marginTop": "24px",
          "minHeight": "200px",
          "paddingTop": "72px",
          "textAlign": "center",
        }
      }
    >
      <div>
        <p>
          Name: 
          test
        </p>
        <p>
          Start releaseDate:
           
          1970-01-01
        </p>
      </div>
    </div>
    <div
      style={
        Object {
          "marginTop": "24px",
        }
      }
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
          }
        }
      >
        <Button
          block={false}
          ghost={false}
          loading={false}
          onClick={[Function]}
          prefixCls="ant-btn"
          type="primary"
        >
          Submit
        </Button>
        <Button
          block={false}
          ghost={false}
          loading={false}
          onClick={[Function]}
          prefixCls="ant-btn"
          style={
            Object {
              "marginLeft": 8,
            }
          }
        >
          Previous
        </Button>
      </div>
    </div>
  </div>
</Adapter>
`;

exports[`Create Edit Course Steps Unit Tests: Create Step 3: Error 1`] = `
<Adapter
  style={
    Object {
      "padding": "0px 24px",
    }
  }
>
  <div
    style={
      Object {
        "background": "#fff",
        "padding": "24px",
      }
    }
  >
    <Steps
      current={2}
      iconPrefix="ant"
      prefixCls="ant-steps"
      status="error"
    >
      <Step
        key="Configure"
        title="Configure"
      />
      <Step
        key="Review"
        title="Review"
      />
      <Step
        key="Finished"
        title="Finished"
      />
    </Steps>
    <div
      style={
        Object {
          "backgroundColor": "#fafafa",
          "border": "1px dashed #e9e9e9",
          "borderRadius": "6px",
          "marginTop": "24px",
          "minHeight": "200px",
          "paddingTop": "72px",
          "textAlign": "center",
        }
      }
    >
      <div>
        <div>
          Cannot Process the Request
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "marginTop": "24px",
        }
      }
    >
      <div
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <Button
          block={false}
          ghost={false}
          loading={false}
          prefixCls="ant-btn"
          type="primary"
        >
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/"
          >
            Back to Course Library
          </Link>
        </Button>
      </div>
    </div>
  </div>
</Adapter>
`;

exports[`Create Edit Course Steps Unit Tests: Create Step 3: No Error 1`] = `
<Adapter
  style={
    Object {
      "padding": "0px 24px",
    }
  }
>
  <div
    style={
      Object {
        "background": "#fff",
        "padding": "24px",
      }
    }
  >
    <Steps
      current={2}
      iconPrefix="ant"
      prefixCls="ant-steps"
      status="finish"
    >
      <Step
        key="Configure"
        title="Configure"
      />
      <Step
        key="Review"
        title="Review"
      />
      <Step
        key="Finished"
        title="Finished"
      />
    </Steps>
    <div
      style={
        Object {
          "backgroundColor": "#fafafa",
          "border": "1px dashed #e9e9e9",
          "borderRadius": "6px",
          "marginTop": "24px",
          "minHeight": "200px",
          "paddingTop": "72px",
          "textAlign": "center",
        }
      }
    >
      <div>
        <div>
          Request Complete
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "marginTop": "24px",
        }
      }
    >
      <div
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <Button
          block={false}
          ghost={false}
          loading={false}
          prefixCls="ant-btn"
          type="primary"
        >
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/"
          >
            Back to Course Library
          </Link>
        </Button>
      </div>
    </div>
  </div>
</Adapter>
`;

exports[`Create Edit Course Steps Unit Tests: Edit Check States 1`] = `
<Adapter
  style={
    Object {
      "padding": "0px 24px",
    }
  }
>
  <div
    style={
      Object {
        "background": "#fff",
        "padding": "24px",
      }
    }
  >
    <Steps
      current={0}
      iconPrefix="ant"
      prefixCls="ant-steps"
      status="wait"
    >
      <Step
        key="Configure"
        title="Configure"
      />
      <Step
        key="Review"
        title="Review"
      />
      <Step
        key="Finished"
        title="Finished"
      />
    </Steps>
    <div
      style={
        Object {
          "backgroundColor": "#fafafa",
          "border": "1px dashed #e9e9e9",
          "borderRadius": "6px",
          "marginTop": "24px",
          "minHeight": "200px",
          "paddingTop": "72px",
          "textAlign": "center",
        }
      }
    >
      <Form(CreateEditCourseForm)
        name={
          Object {
            "value": "name",
          }
        }
        onChange={[Function]}
        onSubmit={[Function]}
        releaseDate={
          Object {
            "value": "1970-01-01T16:00:00.000Z",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "marginTop": "24px",
        }
      }
    />
  </div>
</Adapter>
`;
